source/operators/frame_extractor/frame_extractor.py

```diff
@@ -0,0 +1,162 @@
+###############################################################################
+# PURPOSE:
+#   Lambda function to extract frames from video files using cv2 and store on S3.
+#   WARNING: This function might needs longer Lambda timeouts depending on chunk 
+#   size and how many frames should be extracted.
+###############################################################################
+
+import cv2
+import base64
+import json
+import os
+import boto3
+from MediaInsightsEngineLambdaHelper import MediaInsightsOperationHelper
+from MediaInsightsEngineLambdaHelper import MasExecutionError
+from MediaInsightsEngineLambdaHelper import DataPlane
+
+# Environment Variables
+store_frames=os.environ.get("STORE_FRAMES", "None")
+frame_resize_width=int(os.environ.get("FRAME_RESIZE_WIDTH", 480 ))
+frame_resize_height=int(os.environ.get("FRAME_RESIZE_HEIGHT", 320 ))
+
+s3 = boto3.client('s3')
+
+def download_chunk(s3bucket, s3key):
+    temp_chunk = '/tmp/temp_chunk'
+    with open(temp_chunk, 'wb') as f:
+        s3.download_fileobj(s3bucket, s3key, f)
+    f.close()
+    return temp_chunk
+
+def put_record_s3(s3bucket, asset_id, chunk_id, frames):
+    file_name = 'private/assets/{}/output/{}/frame_list.json'.format(asset_id, chunk_id)
+    try:
+        s3.put_object( 
+            ACL='bucket-owner-full-control', 
+            Bucket=s3bucket, 
+            Key=file_name, 
+            Body=json.dumps({"chunk": frames}).encode()
+        )
+    except Exception as e:
+        print("put_record_s3: Failed saving file %s" % file_name)
+        print (e)
+
+def lambda_handler(event, context):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 85 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.